### PR TITLE
SLSKPROTOCOL.md: remove 'Function Names' sections

### PR DESCRIPTION
**Pull request by slook**
- Removed: All "Function Names" sub-sections, because all of them contained only 1 item
- Changed: All "Description" sub-section headers to the CamelCase name of the function used in Nicotine+
- Added: Some missing links for easier cross-referencing, added missing descriptions for wishlist related server messages
- Changed: File Attribute Types to prioritize official SoulseekQt latest specs, and add Soulseek.NET and Seeker.

Before: image - - - - After: image

This also has the advantage of being able to link to `#FunctionName` if desired.

### CODE REVIEW

File: doc/SLSKPROTOCOL.md [hunk 1]
```diff
@@ -67,25 +67,25 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
+| ---- | ----------------- |
+| 0    | Bitrate (kbps)    |
+| 1    | Length (seconds)  |
+| 2    | VBR (0 or 1)      |
+| 3    | Encoder (unused)  |
+| 4    | Sample Rate (Hz)  |
+| 5    | Bit Depth (bits)  |
 
-#### Used Attribute Combinations
+#### File Attribute Combinations
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:88]: The title needs to emphasize that these combinations are actively used by clients. More combinations are possible, of course.

File: doc/SLSKPROTOCOL.md [hunk 2]
```diff
@@ -229,17 +229,15 @@ and callbacks for the messages are set in pynicotine.py.
 | 1001 | [Can't Connect To Peer](#server-code-1001)        |            |
 | 1003 | [Can't Create Room](#server-code-1003)            |            |
 
+The Nicotine+ function names used in slskmessages.py are specified below.
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:232]: Not needed, imo.

File: doc/SLSKPROTOCOL.md [hunk 3]
```diff
@@ -128,9 +128,9 @@ and callbacks for the messages are set in pynicotine.py.
 | -------------- | ------- | ---------------- |
 | uint32         | uint32  | ...              |
 
-### Message Index
+### Server Message Codes
 
-| Code | Message                                           | Status     |
+| Code | Message Function                                  | Status     |
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:133]: I prefer "Message".

File: doc/SLSKPROTOCOL.md [hunk 4]
```diff
@@ -258,7 +256,7 @@ We send this to the server right after the connection has been established. Serv
 00 64 35 31 63 39 61 37 65 39 33 35 33 37 34 36 61 36 30 32 30 66 39 36
 30 32 64 34 35 32 39 32 39 01 00 00 00**
 
-### Data Order
+#### Data Order
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:259]: Could you revert to the old heading level for these (this one looks too similar to regular bold text)?

**Comment by mathiascode** [comment]
Any changes in the descriptions here should also be added to slskmessages.py.

File: doc/SLSKPROTOCOL.md (hunk 1 above)
Comment (slook) [reply doc/SLSKPROTOCOL.md:88]: "`Known`" ?

File: doc/SLSKPROTOCOL.md (hunk 4 above)
Comment (slook) [reply doc/SLSKPROTOCOL.md]: I've gone for _italics_ instead of a tag, because tagging these sub-headings do not serve any useful purpose for navigation.

**Comment by slook** [comment]
Done.

I have changed the chronological order of the File Attribute Combinations section, let me know if you don't approve?

File: doc/SLSKPROTOCOL.md [hunk 5]
```diff
@@ -67,31 +67,27 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
 | 0    | Bitrate (kbps)   |
 | 1    | Length (seconds) |
 | 2    | VBR (0 or 1)     |
 | 3    | Encoder (unused) |
 | 4    | Sample Rate (Hz) |
-| 5    | Bit Depth        |
+| 5    | Bit Depth (bits) |
 
-#### Used Attribute Combinations
+*File Attribute Combinations:*
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:88]: Can't be linked to, please revert to a heading.

File: doc/SLSKPROTOCOL.md [hunk 6]
```diff
@@ -67,31 +67,27 @@ If you find any inconsistencies, errors or omissions in the documentation, pleas
...
 
-  - Soulseek NS, SoulseekQt (2015-2-21 and earlier), Nicotine+, Museek+, SoulSeeX, slskd (lossy formats):
-      - {0: *bitrate*, 1: *length*, 2: *VBR*}
-
-  - SoulseekQt (2015-2-21 and earlier):
-      - {0: *bitrate*, 2: *VBR*}
+These combinations are actively used by clients. More combinations are possible, of course.
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:90]: We shouldn't encourage new combinations unless the official client makes changes.

File: doc/SLSKPROTOCOL.md [hunk 7]
```diff
@@ -231,64 +235,53 @@ and callbacks for the messages are set in pynicotine.py.
...
-
-  - Nicotine+: Login
-
-### Sending Login Example
-
-| Description | Message Length | Message Code | Username Length | Username                | Password Length | Password                |
-| ----------- | -------------- | ------------ | --------------- | ----------------------- | --------------- | ----------------------- |
-| Human       | 72             | 1            | 8               | username                | 8               | password                |
-| Hex         | 48 00 00 00    | 01 00 00 00  | 08 00 00 00     | 75 73 65 72 6e 61 6d 65 | 08 00 00 00     | 70 61 73 73 77 6f 72 64 |
+#### Sending Login Example
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:240]: Revert to a level 3 heading, this was already changed once.

File: doc/SLSKPROTOCOL.md [hunk 8]
```diff
@@ -231,64 +235,53 @@ and callbacks for the messages are set in pynicotine.py.
...
-00 64 35 31 63 39 61 37 65 39 33 35 33 37 34 36 61 36 30 32 30 66 39 36
-30 32 64 34 35 32 39 32 39 01 00 00 00**
+*Message as a Hex Stream:*
+  - **48 00 00 00 01 00 00 00 08 00 00 00 75 73 65 72 6e 61 6d 65 08 00 00 00 70 61 73 73 77 6f 72 64 a0 00 00 00 20 00
+      00 00 64 35 31 63 39 61 37 65 39 33 35 33 37 34 36 61 36 30 32 30 66 39 36 30 32 64 34 35 32 39 32 39 01 00 00 00**
 
-### Data Order
-
-  - Send
+*Data Order:*
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:258]: Visually and semantically, I still prefer a heading here. Please revert to a level 3 heading, as these are all sections under `Server Code 1`.

File: doc/SLSKPROTOCOL.md [hunk 9]
```diff
@@ -231,64 +235,53 @@ and callbacks for the messages are set in pynicotine.py.
...
+    4.  **string** <ins>hash</ins> *MD5 hex digest of concatenated username & password*
+    5.  **uint32** <ins>minor version</ins> **0x13000000** for 157 ns 13e, **0x11000000** for 157 ns 13c
   - Receive Login Success
-    1.  **bool** <ins>success</ins> 1
-    2.  **string** <ins>greet</ins> A MOTD string
+    1.  **bool** <ins>success</ins> **1**
+    2.  **string** <ins>greet</ins> *MOTD string*
     3.  **uint32** <ins>Your IP Address</ins>
-    4.  **string** <ins>MD5 hex digest of the password string</ins> *Windows Soulseek uses this hash to determine if it's connected to the official server*
+    4.  **string** <ins>hash</ins> *MD5 hex digest of the password string (Windows SoulseekQt uses this hash to determine if it's connected to the official server)*
```
Comment (mathiascode) [inline doc/SLSKPROTOCOL.md:269]: This statement is probably not relevant anymore, and should be removed. Windows Soulseek refers to Soulseek NS before the new server was introduced.

**Comment by mathiascode** [comment]
Could you also check the message index? Some messages are still using the old Museek+ message names, like server code 150 (Ask Public Chat).

**Comment by mathiascode** [comment]
> Could you also check the message index? Some messages are still using the old Museek+ message names, like server code 150 (Ask Public Chat).

This can be done later, I'll merge the PR now. Thanks!

**Comment by slook** [comment]
Maybe the message index requires further thought, the English titles are more readable but are not technically correct as to the actual function names. Perhaps a short Description column would be helpful, but then there would be too much redundant info.

File: doc/SLSKPROTOCOL.md (hunk 8 above)
Comment (slook) [reply doc/SLSKPROTOCOL.md]: Perhaps, the "Data Order" headings are not needed at all?

**Pull request merged**